common/helpers.js

```diff
@@ -2,24 +2,47 @@ const url = require('url');
 const config = require('../config.json');
 
 const dataGuidsHostPrefix = 'dg.';
-const dataObjectPathPrefix = '/ga4gh/dos/v1/dataobjects/';
+const dosDataObjectPathPrefix = '/ga4gh/dos/v1/dataobjects/';
+const drsDataObjectPathPrefix = '/ga4gh/drs/v1/objects/';
+const jadeDataRepoHostRegex = /jade(-terra)?.datarepo-(dev|prod).broadinstitute.org/;
+
+// Regex drops any leading or trailing "/" characters and gives the path out of capture group 1
+const pathSlashRegex = /^\/?([^/]+.*?)\/?$/;
+
+const samBaseUrl = () => config.samBaseUrl;
+const jadeDataRepoUrl = () => config.jadeDataRepoBaseUrl;
+
 
 function hasDataGuidsHost(someUrl) {
     return someUrl.host.startsWith(dataGuidsHostPrefix);
 }
 
+function hasJadeDataRepoHost(someUrl) {
+    return someUrl.host === jadeDataRepoUrl();
+}
+
 function isDataGuidsUrl(someUrl) {
     return hasDataGuidsHost(someUrl) || (someUrl.hostname && !someUrl.pathname);
 }
 
 function validateDataObjectUrl(someUrl) {
-    if (hasDataGuidsHost(someUrl) && !someUrl.pathname) {
-        throw new Error(`Data Object URIs with '${dataGuidsHostPrefix}*' host are required to have a path: "${url.format(someUrl)}"`);
+    const hasJDRHost = hasJadeDataRepoHost(someUrl);
+
+    if ((hasDataGuidsHost(someUrl) || hasJDRHost) && !someUrl.pathname) {
+        throw new Error(`Data Object URIs with either '${dataGuidsHostPrefix}*' or '${jadeDataRepoUrl()}' as host are required to have a path: "${url.format(someUrl)}"`);
```

Comment:
Similarly, I would just mention that the url is a type of JDR URL, as there will definitely be more JDRs beyond the `terra` one.